### PR TITLE
Refactor Authentication + Sidebar Navigation

### PR DESCRIPTION
**Pull request by chansen424**
### Summary 
Added Navigation via a Sidebar menu. Refactored Authentication so that the logout button could be placed in a different location than the sign in button, and refreshing does not log the user out. Minor refactors in other files so that site works with backend updates.

This pull request is the first step towards implementing the Carriage Admin Dashboard

- [x] Refactored Authentication
- [x] Routing
- [x] Sidebar

### Test Plan 
Login. Navigate between the pages, making sure that the menu item selected, the url, and the content being rendered on the right-hand side of the page are consistent. Refresh on any page other than the dashboard home to ensure that refreshing leads you back to the same page. Logout to ensure that you are returned to the landing page.

### Notes 
Ignore the grey bar at the top of the landing page, since it will be dealt with once the design of the landing page is solidified.

### CODE REVIEW

File: frontend/src/components/AuthManager.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import { GoogleLogin } from 'react-google-login';
+import { useHistory, useLocation } from "react-router-dom";
+import useClientId from '../hooks/useClientId';
+import AuthContext from '../context/auth';
+import LandingPage from './LandingPage';
+
+export const AuthManager: FunctionComponent = ({ children }) => {
+  const [signedIn, setSignedIn] = useState(false);
+  const clientId = useClientId();
+  let history = useHistory();
```
Comment (mg2239) [inline frontend/src/components/AuthManager.tsx:11]: history isnt being reassigned so i would use const

File: frontend/src/components/AuthManager.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  function logout() {
+    localStorage.clear();
+    setSignedIn(false);
+    history.push('/');
+  }
+
+  async function onSignIn(googleUser: any) {
+    let token = googleUser.getAuthResponse().id_token;
+    let email = googleUser.profileObj.email;
```
Comment (mg2239) [inline frontend/src/components/AuthManager.tsx:22]: again using const here, and i think destructuring would look cleaner
```
const { id_token: token } = googleUser.getAuthResponse();
const { email } = googleUser.profileObj;
// or even const { profileObj: { email } } = googleUser;
```

File: frontend/src/components/AuthManager.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          {children}
+        </AuthContext.Provider>
+      )
+      : (
+        <>
+          <GoogleLogin
+            onSuccess={onSignIn}
+            onFailure={() => console.error("failed to sign in")}
+            clientId={clientId}
+            cookiePolicy={'single_host_origin'}
```
Comment (mg2239) [inline frontend/src/components/AuthManager.tsx:63]: don't need curly braces here

File: frontend/src/components/Sidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  const [selected, setSelected] = useState(menuItems.findIndex((item) => item.path === pathname));
+
+  return (
+    <div className="container">
+      <div>
+        <div className="sidebar">
+          {menuItems.map((item, index) => (
+            <Link key={item.path} className="sidebar-links" to={item.path}>
+              <div onClick={() => setSelected(index)} className="icon">
+                <div className={index === selected ? "selected circle" : "circle"}>
```
Comment (mg2239) [inline frontend/src/components/Sidebar.tsx:33]: you can just set the initial state of `selected` to be `pathname`, then change `setSelected(index)` to `setSelected(item.path)` and `index === selected` to `item.path === selected`

File: frontend/src/pages/RidersTable.tsx [hunk 5]
```diff
@@ -265,9 +265,12 @@ const Table = () => {
...
   useEffect(() => {
     async function getExistingRiders() {
-      const response = await fetch('/riders');
-      const data = (await response.json());
-      let allRiders: Rider[] = data.map(function (rider: any) {
+      const ridersData = await fetch('/riders').then(res => {
+        return res.json();
+      }).then(data => {
+        return data["data"];
+      });
```
Comment (mg2239) [inline frontend/src/pages/RidersTable.tsx:272]: rewrite to be single line arrow functions, and either dot notation or destructure the parameter for data

File: frontend/src/components/AuthManager.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      setSignedIn(true);
+      if (pathname === '/') {
+        history.push('/dashboard/home');
+      }
+    } else {
+      logout();
+    }
+  }
+
+  return (
```
Comment (mg2239) [inline frontend/src/components/AuthManager.tsx:50]: i would personally use inline ifs here, but is there an advantage to using a ternary statement?

File: frontend/src/components/Sidebar.tsx (hunk 4 above)
Comment (chansen424) [reply frontend/src/components/Sidebar.tsx:33]: Really nice catch 👍

File: frontend/src/components/AuthManager.tsx (hunk 2 above)
Comment (chansen424) [reply frontend/src/components/AuthManager.tsx:22]: Nice suggestion. Should have cleaned up more, but a lot of this was copy-pasted from Lucy's original component.

File: frontend/src/components/AuthManager.tsx (hunk 6 above)
Comment (chansen424) [reply frontend/src/components/AuthManager.tsx]: Cleaned it up a bit, but just thought that the ternary really enforces that this is an if-else situation. Probably would have used && in situations where the ternary would be `variable ? something : <></>` to eliminate the unnecessary fragment.

Comment (mg2239) [reply frontend/src/components/AuthManager.tsx]: yeah i got what youre saying, makes sense to me 👍

File: frontend/src/components/Sidebar.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  const [selected, setSelected] = useState(pathname);
+
+  const menuItems: MenuItem[] = [{ icon: home, caption: "Home", path: "/home" },
+  { icon: drivers, caption: "Drivers", path: "/drivers" },
+  { icon: riders, caption: "Riders", path: "/riders" },
+  { icon: settings, caption: "Settings", path: "/settings" }];
+
+  return (
+    <div className="container">
+      <div>
```
Comment (mg2239) [inline frontend/src/components/Sidebar.tsx:27]: remove unnecessary div

**Review by mg2239 — CHANGES_REQUESTED** [review]

**Review by mg2239 — APPROVED** [review]

**Pull request merged**